freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/RepairInstancesService.java

```diff
@@ -255,4 +267,25 @@ public OperationStatus rebootInstances(String accountId, RebootInstancesRequest
         }
         return operationToOperationStatusConverter.convert(operation);
     }
+
+    public DescribeFreeIpaResponse rebuild(String accountId, RebuildRequest rebuildRequest) {
+        ValidationResult.ValidationResultBuilder validationResultBuilder = ValidationResult.builder();
+        Optional<Stack> stack =
+                stackService.findByCrnAndAccountIdWithListsEvenIfTerminated(rebuildRequest.getEnvironmentCrn(), accountId, rebuildRequest.getSourceCrn());
+        if (stack.isEmpty()) {
+            validationResultBuilder.error("Stack not found.");
+        }
+        Optional<Stack> nonTerminatedStack = stackService.findByEnvironmentCrnAndAccountId(rebuildRequest.getEnvironmentCrn(), accountId);
```

Comment:
what if the stack is terminated on the provider side? It would still show up here, so I think it wouldn't resolve the case what most probably our customers would face, as FreeIPA termination is not available on the public API, only environment termination is. If the environment is terminated I don't think FreeIPA could be rebuilt.

Reply:
You are correct. We will need to expose more APIs in the future. Right now, for the hotfix we are going to use curl to directly invoke both the delete API and the rebuild API.

Reply:
ok, just please don't forget to file a jira for this